### PR TITLE
ga 5.3 (new formula)

### PR DESCRIPTION
**Pull request by pjknowles**
Add the Global Arrays (GA) toolkit that provides an efficient and portable "shared-memory" programming interface for distributed-memory computers.

### CODE REVIEW

File: Library/Formula/ga.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+class Ga < Formula
+  homepage "http://hpc.pnl.gov/globalarrays/"
+  url "http://hpc.pnl.gov/globalarrays/download/ga-5-3.tgz"
+  version "5.3"
+  sha1 "b4e5fdc7f25e76d4d9e4c8b23f9c9ae2855cd925"
+
+  depends_on "gcc"
```
Comment (mikemcquaid) [inline Library/Formula/ga.rb]: Use `fails_with` here instead. Note that we'll probably not include this if it doesn't build with Clang.

File: Library/Formula/ga.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+           "F77=mpif77",
+           "CXX=mpicxx",
+           "MPICC=mpicc",
+           "MPIFC=mpif90",
+           "MPIF77=mpif77",
+           "MPICXX=mpicxx",
+           "--enable-cxx",
+           "--prefix=#{prefix}"
+    system "make"
+    #system "make check MPIEXEC='mpiexec -np 2'"
```
Comment (mikemcquaid) [inline Library/Formula/ga.rb]: Either uncomment or delete this.

File: Library/Formula/ga.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        testpattern=pattern;
+        GA_Brdcst(&testpattern,sizeof(testpattern),0);
+        printf("[%d/%d] Hello, world! My name is %s; broadcast error=%d.\\n", rank, size, name,testpattern-pattern);
+        GA_Terminate();
+        MPI_Finalize();
+        return testpattern != pattern;
+      }
+    EOS
+    ldflags = `#{bin}/ga-config --ldflags --libs`
+    system("mpicc hello.c -o hello "+ldflags)
```
Comment (mikemcquaid) [inline Library/Formula/ga.rb]: `system "mpicc", "hello.c". "-o", "hello", *ldflags.split`

**Comment by mikemcquaid** [comment]
Would this make more sense in Homebrew Science?

**Comment by mikemcquaid** [comment]
`brew audit --strict` and `brew install` are failing: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/21279/version=yosemite/testReport/

**Comment by pjknowles** [comment]
My apologies that I have probably wasted some of your time with this. I have realised that the contortions that I coded - which resulted from carrying in a shell script into the brew world - are not portable and not the best thing.  It is best that I keep this private for the moment till it is sorted out solidly.

There is a generic issue that I am not sure whether brew can really support. Package A (ga) requires package B (mpich2) to build it, but it works only if B has been built with gcc not clang, since B provides the compiler for the build of A, and A has not been properly ported to clang by its owners.

I have thought of another way to do it, which is to have my own tap containing both formulae, so a copy of mpich2 with 'fails_with :clang' added, but this is not ideal since it has to be maintained in parallel with the official mpich2 formula.


> On 12 Feb 2015, at 19:18, Mike McQuaid <notifications@github.com> wrote:
> 
> Would this make more sense in Homebrew Science?
> 
> —
> Reply to this email directly or view it on GitHub.
> 

Peter Knowles
Rhosgoch, Cefn Bychan, Pentyrch CF15 9PL, UK
Peter@Knowles-family.me.uk
07906 029489

**Comment by mikemcquaid** [comment]
Oh, no need to apologise at all here, I appreciate your work here and sorry it couldn't get included.

**Pull request closed without merging**